### ISSUE TITLE
Provide dense version of inline notification

### ISSUE THREAD

**Issue by huwshimi** [canonical-web-and-design/vanilla-framework#3970]
## Pattern to amend

Notification

## Visual

<img width="411" alt="Screen Shot 2021-09-06 at 2 45 22 pm" src="https://user-images.githubusercontent.com/361637/132162133-e363cad2-d36b-4b4e-8724-269a6d384092.png">

## Context

While trying to implement the visual above I first tried the following:

```html
<div class="p-notification--caution is-borderless is-inline">
  <div class="p-notification__content">
    <p class="p-notification__message">Subscription expired</p>
  </div>
</div>
```
This had too much spacing above, below and between the icon and the text.

<img width="414" alt="Screen Shot 2021-09-06 at 2 46 03 pm" src="https://user-images.githubusercontent.com/361637/132162336-73a5c8c2-b7b3-42b9-93df-723f2159b9c9.png">

After adding `u-no-margin--bottom` that corrected the bottom spacing, but there wasn't away to correct the rest of the spacing.

<img width="418" alt="Screen Shot 2021-09-06 at 2 46 20 pm" src="https://user-images.githubusercontent.com/361637/132162348-84978992-421e-4eaa-af4d-1adfee3e5984.png">

**Comment by huwshimi** [comment]
Using the following appears to provide the correct spacing:

```scss
  .is-dense {
    background-position-y: center;
    margin-bottom: 0;
    padding-left: $sph-inner--large;

    .p-notification__content {
      padding-top: 0;
    }
  }
```

**Comment by lyubomir-popov** [comment]
Hi Hew, not sure how the non-dense versions were implemented, buyt they are not what the stock notifications look like - here's what it should have looked like, and I'd argue there's no need for a denser version than thihs:

![image](https://user-images.githubusercontent.com/2741678/132184530-a8af8c4d-d74c-4cec-a5d7-c9733467c947.png)

Additional notes:
- the notification design doe not allow for samll caps labels flush right - information like this can be reformatted to become part of the main copy. The notification is not an agnostic container that can be filled with any preformatted content. 
- The first indented line next to thecheckbox sets a "keyline" - any text underneath should align to that first line, not the icon
- The title should be bold, otherwise there's a prominence conflict - title here being regular body text, and the "Free personal token" trying to upstage it by being bold 
- As far as I know, we use sentence case, and I see title case here 
- If the design insists on using the pre-formatted version of the plans, then it should sit under the notification, not inside it. 

I suggest we close this as this is improper use of the component and there are ways to achieve the same result without the need for a new flavour of the notification.

**Comment by huwshimi** [comment]
Thanks @lyubomir-popov, I should have given a bit more context of the design. This is a card that sometimes has a notification at the top. It has a selected state which gives it that orange left border, here's what the full card looks like in the unselected state:

<img width="414" alt="Screen Shot 2021-09-06 at 7 13 50 pm" src="https://user-images.githubusercontent.com/361637/132192244-8fd19ddd-982f-45cb-9c96-47bfe5967e6b.png">


The full designs can be seen [here](https://app.zeplin.io/project/60ec5fb4e07cc90fbbf2b318/screen/60f02ee64591b069a6f5dc0c).

**Comment by lyubomir-popov** [comment]
I see. I still don't think a new variant is needed. The problem here arises because you have less spacing vertically between less related elements (notification content, title underneath) than between the coupled title and "yearly" label. I would propose a hr in between, which creates a clearer picture of what belongs where:

![image](https://user-images.githubusercontent.com/2741678/132215535-4d5b8c2b-2e96-4a49-bc24-f8ff6d4e1e56.png)

This way, the spacing between icon and notification copy doesn't interfere with the rest of the layout, as it is in a row of its own.

**Comment by lyubomir-popov** [comment]
@anasereijo do you agree with the above? If so we can maybe close this?

**Comment by anasereijo** [comment]
@lyubomir-popov I like your proposal by using the hr, thanks for that. I think we can then close this.

**Comment by huwshimi** [comment]
This actually creates more custom code as it needs to remove padding from the card and reapply it to the child elements (with a wrapping element to apply the correct padding for the notification as it already uses padding to leave space for the icon and another wrapping element to apply padding for the card content after the `<hr>`).

I don't mind which custom code we go with. @anasereijo should I leave it as it is or should I add the hr?

**Comment by lyubomir-popov** [comment]
@huwshimi this pr will make it easier (at least in plan vanilla) to create elements that touch the card's container.  Maybe wai until this is released so you don't have to write local code? https://github.com/canonical-web-and-design/vanilla-framework/pull/3953

**Comment by huwshimi** [comment]
> @huwshimi this [pr](https://github.com/canonical-web-and-design/vanilla-framework/pull/3953) will make it easier to create elements that touch the card's container. Maybe wai until this is released so you don't have to write local code? @sowasred2012 this will be released in a week or so right?

Thanks, that definitely looks like it would make things easier. I'll still have to add a wrapping element around the notification to 
be able to provide padding on either side:

<img width="434" alt="Screen Shot 2021-09-08 at 4 54 49 pm" src="https://user-images.githubusercontent.com/361637/132461304-0127723a-efa9-4de9-9dad-3df5e7990d84.png">

**Comment by lyubomir-popov** [comment]
keep the notificaiton in `.p-card__inner`, then insert an `<hr>` underneath. CC @sowasred2012

**Comment by huwshimi** [comment]
> @huwshimi keep the notificaiton in `.p-card__inner`, then insert an `<hr>` underneath. CC @sowasred2012

But then the hr will get padding on either side...

**Comment by lyubomir-popov** [comment]
.p-card__inner applies the padding, not p-card. that's exactly for cases like these.

**Comment by huwshimi** [comment]
> In the new version of the card, `.p-card__inner` applies the padding, not p-card. that's exactly for cases like these. right @sowasred2012 ?

Ah right, so you're saying have two `p-card_inner`s? So you'd end up with something like this?

```html
<div class="p-card">
  <div class="p-card__inner u-no-padding--top u-no-padding--bottom">
    ....notification
  </div>
  <hr />
  <div class="p-card__inner u-no-padding--top">
   ...card content
  </div>
</div>
```

**Comment by lyubomir-popov** [comment]
yes, as many as you need. It's just a way to padd things from one side or both (with u-no-padding.. if you need just one side)

**Comment by huwshimi** [comment]
> yes, as many as you need. It's just a way to pad things from one side or both (with u-no-padding.. if you need just one side)

Thanks, that sounds good. I'll wait for that release then.

**Comment by anasereijo** [comment]
@huwshimi sorry I thought @lyubomir-popov's solution would be something already implemented on vanilla. I was about to suggest keeping it the same, without the hr, but from what I see above you guys already figured it out? Let me know if you need anything else from my side.

**Comment by huwshimi** [comment]
> @huwshimi sorry I thought @lyubomir-popov's solution would be something already implemented on vanilla. I was about to suggest keeping it the same, without the hr, but from what I see above you guys already figured it out? Let me know if you need anything else from my side.

Yep sorry a lot of messages, I think I'll be able to do the hr when the next Vanilla release happens.

**Comment by sowasred2012** [comment]
@huwshimi @lyubomir-popov sorry, missed all the action! Yep, the `p-card__inner` PR is merged now, so it'll be available with the next release: 2.36.0.
